File: android/app/src/main/kotlin/com/enolo/app/data/dto/SommelierDto.kt
package com.enolo.app.data.dto

import kotlinx.serialization.Serializable

@Serializable
data class ChatWinePickDto(
    val cellarItemId: String,
    val title: String,
    val reason: String = "",
)

@Serializable
data class ChatSourceDto(
    val book: String,
    val page: Int? = null,
    val heading: String? = null,
)

@Serializable
data class ChatMessageDto(
    val id: String,
    val role: String, // "user" | "assistant"
    val mode: String? = null, // "pogreb" | "consult" | "chat"
    val content: String,
    val picks: List<ChatWinePickDto> = emptyList(),
    val sources: List<ChatSourceDto> = emptyList(),
    val createdAt: String = "",
)

@Serializable
data class ChatSessionDto(
    val id: String,
    val title: String? = null,
    val createdAt: String = "",
    val updatedAt: String = "",
)

@Serializable
data class ChatSessionWithMessagesDto(
    val id: String,
    val title: String? = null,
    val createdAt: String = "",
    val updatedAt: String = "",
    val messages: List<ChatMessageDto> = emptyList(),
)

@Serializable
data class SendMessageRequest(val text: String)

/** Событие потоковой выдачи ответа сомелье (NDJSON-строки от стрим-эндпоинта). */
sealed class ChatStreamEvent {
    data class Delta(val text: String) : ChatStreamEvent()
    data class Done(val message: ChatMessageDto) : ChatStreamEvent()
    data class Error(val message: String) : ChatStreamEvent()
}

File: code/src/modules/sommelier/sommelier.dto.ts
/** DTO раздела «AI-сомелье» (разговорный чат поверх погреба и винной базы знаний). */

export interface SendMessageDto {
  /** Текст пользователя. Может начинаться со слэш-команды: «/погреб …» или «/консультация …». */
  text: string
}

/** Карточка вина из погреба (для режима /погреб). */
export interface ChatWinePick {
  cellarItemId: string
  title: string
  reason: string
}

/** Цитата-источник из винной литературы. */
export interface ChatSource {
  book: string
  page: number | null
  heading: string | null
}

export interface ChatMessageView {
  id: string
  role: 'user' | 'assistant'
  mode: 'pogreb' | 'consult' | 'chat' | null
  content: string
  picks: ChatWinePick[]
  sources: ChatSource[]
  createdAt: string
}

export interface ChatSessionView {
  id: string
  title: string | null
  createdAt: string
  updatedAt: string
}

export interface ChatSessionWithMessages extends ChatSessionView {
  messages: ChatMessageView[]
}
